library/common-library/templates/_secret.yaml.tpl

```diff
@@ -0,0 +1,87 @@
+{{/*
+Return local licenseKey if set, global otherwise.
+This helper is for internal use.
+*/}}
+{{- define "common.licenseKey._licenseKey" -}}
+{{- if .Values.licenseKey -}}
+  {{- .Values.licenseKey -}}
+{{- else if .Values.global -}}
+  {{- if .Values.global.licenseKey -}}
+    {{- .Values.global.licenseKey -}}
+  {{- end -}}
+{{- end -}}
+{{- end -}}
+
+{{/*
+Return the name of the secret holding the License Key.
+This helper is for internal use.
+*/}}
+{{- define "common.licenseKey._customSecretName" -}}
+{{- if .Values.customSecretName -}}
+  {{- .Values.customSecretName -}}
+{{- else if and .Values.global -}}
+  {{- if .Values.global.customSecretName -}}
+    {{- .Values.global.customSecretName -}}
+  {{- end -}}
+{{- end -}}
+{{- end -}}
+
+{{/*
+Return the name of the secret holding the License Key.
+*/}}
+{{- define "common.licenseKey.secretName" -}}
+{{ include "common.licenseKey._customSecretName" . | default (printf "%s-license" (include "common.naming.fullname" . )) }}
+{{- end -}}
+
+{{/*
+Return the name key for the License Key inside the secret.
+This helper is for internal use.
+*/}}
+{{- define "common.licenseKey._customSecretKey" -}}
+{{- if .Values.customSecretLicenseKey -}}
+  {{- .Values.customSecretLicenseKey -}}
+{{- else if .Values.global -}}
+  {{- if .Values.global.customSecretLicenseKey }}
+    {{- .Values.global.customSecretLicenseKey -}}
+  {{- end -}}
+{{- end -}}
+{{- end -}}
+
+{{/*
+Return the name key for the License Key inside the secret.
+*/}}
+{{- define "common.licenseKey.secretKeyName" -}}
+{{ include "common.licenseKey._customSecretKey" . | default "licenseKey" }}
+{{- end -}}
+
+{{/*
+Charts using this library can implement this function. If it returns a truty value, common.licenseKey.secret will not
+fail if an empty licenseKey is provided.
+*/}}
+{{- define "common.licenseKey.overrides.allowEmpty" -}}
+{{- end }}
+
+{{/*
+Renders the license key secret if user has not specified a custom secret.
+*/}}
+{{- define "common.licenseKey.secret" }}
+{{- if not (include "common.licenseKey._customSecretName" .) }}
+
+{{/* Fail if licenseKey is empty and required */}}
+{{- if not (include "common.licenseKey._licenseKey" .) }}
+    {{- if not (include "common.licenseKey.overrides.allowEmpty" .) }}
+    {{- fail "You must specify a licenseKey or a customSecretName containing it" }}
+    {{- end }}
+{{- end }}
+
+apiVersion: v1
+kind: Secret
+metadata:
+  name: {{ include "common.licenseKey.secretName" . }}
+  namespace: {{ .Release.Namespace }}
+  labels:
+    {{- include "common.labels" . | nindent 4 }}
+data:
+  {{ include "common.licenseKey.secretKeyName" . }}: {{ include "common.licenseKey._licenseKey" . | b64enc }}
```

Comment:
Therefore if someone is adding the `customSecretLicenseKey ` we render the secret using such value. 

This is something that would make sense? I would expect that either it is custom or it is not.

Reply:
I use the `include` here not to allow users to override this key (as it does not make much sense), but to avoid having the hardcoded `default "licenseKey"` in two places.

Reply:
ok, therefore we allow it, but it is not an issue. Everything should be working in the exact same way